keanu-project/src/main/java/io/improbable/keanu/distributions/discrete/Categorical.java

```diff
@@ -3,10 +3,12 @@
 import java.util.Map;
 
 import io.improbable.keanu.distributions.Distribution;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
 import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.generic.GenericTensor;
 import io.improbable.keanu.vertices.dbl.KeanuRandom;
 
-public class Categorical<T> implements Distribution<T> {
+public class Categorical<T> implements Distribution<GenericTensor<T>> {
```

Comment:
Is there a reason this has to be a `GenericTensor` as opposed to the interface `Tensor`?